src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -38,15 +44,61 @@ public override void Initialize(AnalysisContext context)
             {
                 var compilation = context.Compilation;
                 var symbolType = compilation.GetTypeByMetadataName(s_symbolTypeFullName);
+
                 if (symbolType is null)
                 {
                     return;
                 }
 
                 context.RegisterOperationAction(context => HandleBinaryOperator(in context, symbolType), OperationKind.BinaryOperator);
+
+                var comparerTypes = s_comparerTypeFullName
+                    .Select(comparerTypeFullName => compilation.GetTypeByMetadataName(comparerTypeFullName))
+                    .WhereNotNull()
+                    .ToImmutableArray();
+
+                if (comparerTypes.IsDefaultOrEmpty)
+                {
+                    return;
+                }
+
+                context.RegisterOperationAction(context => HandleOperation(in context, comparerTypes), OperationKind.ObjectCreation);
             });
         }
 
+        private void HandleOperation(in OperationAnalysisContext context, ImmutableArray<INamedTypeSymbol> comparerTypes)
+        {
+            if (context.Operation is IObjectCreationOperation objectCreation)
```

Comment:
Should probably assert or just hard cast. You've registered for object creation, you shouldn't get anything that's not.

Reply:
@ryzngard I marked this comment as unresolved, since the code has not changed and there is no response given

Reply:
For trivial comments that I agree with, I generally mark as resolve once fixed. Is there a better way? 

I just pushed the latest, re-resolving.